datalad/plugin/tests/test_addurls.py

```diff
@@ -446,7 +456,12 @@ def get_annex_commit_counts():
         if not dl_cfg.get('datalad.fake-dates'):
             # We should have two new commits on the git-annex: one for the
             # added urls and one for the added metadata.
-            eq_(n_annex_commits + 2, get_annex_commit_counts())
+            if not on_windows:
+                eq_(n_annex_commits + 2, get_annex_commit_counts())
+            else:
+                # On Windows, there should be one additional "adjusted branch" 
+                # commit for each of the new commits
+                eq_(n_annex_commits + 2 + 2, get_annex_commit_counts())
```

Comment:
Hmm, something seems off here.  If it's about adjusted branches, then this condition should be for that (e.g. `is_managed_branch`) rather than windows.  But when I run this under `./tools/eval_under_testloopfs` (which results in using adjusted branches), the original non-windows code path passes.  I also wasn't aware that using adjusted branches modified the git-annex branch.  Not sure yet what's going on...

Reply:
ah, thanks. I think I must have jumped too quickly to conclusions.
I don't know why the commit count differs, but I can at least paste here *how* it differs. Maybe someone gets an insight based on this.

Here is the difference in the git log that I am seeing between the test repo on a Windows 10 machine and my Debian machine:

- On ``dl-test-branch``:

```
#DEBIAN
adina@muninn in /tmp/datalad_temp_test_addurlsd2vmpg6m on git:dl-test-branch
❱ git log --oneline                                                       
4546b36 (HEAD -> dl-test-branch) [DATALAD] add files from URLs
ad68e47 [DATALAD] new dataset

# WINDOWS
(base) C:\Users\datalad\AppData\Local\Temp\datalad_temp_52vxkqaa>git log --oneline
6c8aba0 (HEAD -> dl-test-branch, refs/basis/adjusted/dl-test-branch(unlocked)) [DATALAD] add files from URLs
c24c000 [DATALAD] new dataset
4453fb6 commit before entering adjusted unlocked branch
```

- the git-annex branch has two extra commits:

```
#DEBIAN 
adina@muninn in /tmp/datalad_temp_test_addurlsd2vmpg6m on git:git-annex
❱ git log --oneline
93f68be (HEAD -> git-annex) update
7ea4803 update
1832b07 update
0629230 branch created

# WINDOWS
(base) C:\Users\datalad\AppData\Local\Temp\datalad_temp_52vxkqaa>git log --oneline
b498e63 (HEAD -> git-annex) update
2eff062 update
311f930 update
d8bf699 update
74d1d99 update
c87051c branch created
```

The differences in the git-annex branches are that the most recent commit on Debian 
```
commit 93f68be159eb43a9222482276a3a49e48d1e9bd6 (HEAD -> git-annex)
Author: DataLad Tester <test@example.com>
Date:   Fri Dec 4 08:32:23 2020 +0100

    update

diff --git a/24f/59a/MD5E-s9--71dd3e865d708f8f8e971309096cd676.log.met b/24f/59>
new file mode 100644
index 0000000..1da66c4
--- /dev/null
+++ b/24f/59a/MD5E-s9--71dd3e865d708f8f8e971309096cd676.log.met
@@ -0,0 +1 @@
+1607067143.327715633s name +b subdir +bar
diff --git a/6ea/286/MD5E-s9--9b72648021b70b8c522642e4490d7ac3.log.met b/6ea/28>
new file mode 100644
index 0000000..9b600f2
--- /dev/null
+++ b/6ea/286/MD5E-s9--9b72648021b70b8c522642e4490d7ac3.log.met
@@ -0,0 +1 @@
+1607067143.418799146s name +c subdir +foo
diff --git a/e21/1c3/MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.log.met b/e21/1c>
new file mode 100644
index 0000000..4687f74
--- /dev/null
+++ b/e21/1c3/MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.log.met
@@ -0,0 +1 @@
+1607067143.185355757s name +a subdir +foo
```

are three separate commits on Windows:
```
(base) C:\Users\datalad\AppData\Local\Temp\datalad_temp_52vxkqaa>git show 311f930
commit 311f93066d969a87ccc2b14f44866f33a29a49c2
Author: DataLad Tester <test@example.com>
Date:   Fri Dec 4 08:18:18 2020 +0100

    update

diff --git a/e21/1c3/MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.log.met b/e21/1c3/MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.log.met
new file mode 100644
index 0000000..589a79d
--- /dev/null
+++ b/e21/1c3/MD5E-s9--3fb7c40c70b0ed19da713bd69ee12014.log.met
@@ -0,0 +1 @@
+1607066297.6131111s name +a subdir +foo

(base) C:\Users\datalad\AppData\Local\Temp\datalad_temp_52vxkqaa>git show 2eff062
commit 2eff062e1390bdac86ce98b71be06fba9b09285e
Author: DataLad Tester <test@example.com>
Date:   Fri Dec 4 08:18:20 2020 +0100

    update

diff --git a/24f/59a/MD5E-s9--71dd3e865d708f8f8e971309096cd676.log.met b/24f/59a/MD5E-s9--71dd3e865d708f8f8e971309096cd676.log.met
new file mode 100644
index 0000000..77a946d
--- /dev/null
+++ b/24f/59a/MD5E-s9--71dd3e865d708f8f8e971309096cd676.log.met
@@ -0,0 +1 @@
+1607066299.4000661s name +b subdir +bar

(base) C:\Users\datalad\AppData\Local\Temp\datalad_temp_52vxkqaa>git show b498e63
commit b498e635e81511ba728a2908b2d4bb7547b08097 (HEAD -> git-annex)
Author: DataLad Tester <test@example.com>
Date:   Fri Dec 4 08:18:21 2020 +0100

    update

diff --git a/6ea/286/MD5E-s9--9b72648021b70b8c522642e4490d7ac3.log.met b/6ea/286/MD5E-s9--9b72648021b70b8c522642e4490d7ac3.log.met
new file mode 100644
index 0000000..e3a419a
--- /dev/null
+++ b/6ea/286/MD5E-s9--9b72648021b70b8c522642e4490d7ac3.log.met
@@ -0,0 +1 @@
+1607066301.1290616s name +c subdir +foo
```

Reply:
In a debugger, on Debian and Windows have the same number of `n_annex_commits`, but  `get_annex_commit_counts()` differ as expected from the git log:
```
# DEBIAN
(Pdb) p get_annex_commit_counts()
4
(Pdb) p n_annex_commits
2

# WINDOWS
(Pdb) p get_annex_commit_counts()
6
(Pdb) p n_annex_commits
2
```

Reply:
Thanks for digging.  So, it looks like you've determined that the metadata commits are the issue.  Before trying to figure out where things are going wrong on our end, let's make sure that git-annex is respecting `alwayscommit=false` on windows.  Can you try (something like) this script on your windows machine and report if the final log command shows the expected output of a single commit?

<details>
<summary>script</summary>

```sh
set -eu

cd "$(mktemp -d "${TMPDIR:-/tmp}"/ga-XXXXXXX)"

alwayscommit=false

git init
git annex init

echo foo >foo
git annex add foo
git -c annex.alwayscommit=$alwayscommit annex metadata --set a=b foo
git -c annex.alwayscommit=$alwayscommit annex metadata --set c=d foo
git commit -mc

set -x
git log --oneline --stat git-annex -- '*.met'
```

</details>

Reply:
I'm getting two commits:

```
+ git log --oneline --stat git-annex -- '*.met'
c44294d (git-annex) update
 ...b9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c.log.met | 1 +
 1 file changed, 1 insertion(+)
2395118 update
 ...b9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c.log.met | 1 +
 1 file changed, 1 insertion(+)
```

Reply:
mhh, but let me make sure this isn't due to some weird bash-on-windows-stuff...

Reply:
Okay, so that looks like a git-annex bug then.  I'd say let's just change the upstream condition to

```
if not (dl_cfg.get('datalad.fake-dates') or on_windows):
```

and extend the comment above that spot to point to this discussion.

I'll open a report on git-annex's side with the information from this thread, though I'll need your help if Joey asks for any follow-up information.

Reply:
> Okay, so that looks like a git-annex bug then. [...] I'll open a report on git-annex's side with the information from this thread, though I'll need your help if Joey asks for any follow-up information.

Oh, what versions of git-annex do you have on your windows machines?  I want to make sure the issue is not actually [a previous one that I reported [1].  That seems unlikely, though, because the issue there was both introduced and fixed after 8.20200330 but before the next release, 8.20200501.

[1] https://git-annex.branchable.com/bugs/commits_created_despite_alwayscommit__61__false_as_of_recent_change/
    43a980829 (disable journal read optimisation when alwayscommit=false, 2020-04-15)

Reply:
(Am currently away from my computer but will check later today :) )

Reply:
The git-annex versions are
```
git-annex version: 8.20201008-g65c1687
```
and 
```
git-annex version: 8.20201008-g7e24b2587
```

Reply:
I've added your proposed change and adjusted the comment. Thanks for the help, @kyleam!

Reply:
Reported: https://git-annex.branchable.com/bugs/windows__58___commits_created_despite_alwayscommit__61__fals/

Reply:
thanks for the report! I have a really stupid, basic question: How could I do a bug report, or comment on an existing bug report?

Reply:
> How could I do a bug report, or comment on an existing bug report?

If you go to <https://git-annex.branchable.com/bugs/>, you'll see a "Report a new bug titled:" box.  After filling in the title and hitting edit, you'll see a login prompt where you can create an account (as well as some other options).  For commenting, when you try to comment, you should also have a chance to log in or create a new account.

Once you file it, it will be committed to the main git-annex repo ([example][1]).  You can select on option to get responses emailed to you.

[1]: https://git.kitenet.net/index.cgi/git-annex.git/commit/?id=d1ffd246bfb42993ff23ec50e01742fb6868bb4a

Reply:
Thank you!!